### ISSUE TITLE
Map not displaying on Chrome on 3 devices

### ISSUE THREAD

**Issue by ramblerswebs** [Leaflet/Leaflet#7699]
Hi
I have no idea if this is the right place to ask for help and I don't know if this is a leaflet bug, chrome or my software.

I am posting hear to see if anyone has an idea of how to track down the problem.

The problem I have is that if a view a specific display using Chrome on one of two laptops or my Android mobile the map does not always display.
If I use a different web browser it works fine. Also if I change the window  width it will display.
I don't know where to turn.

**Steps to reproduce**
Steps to reproduce the behaviour:
- step 1 go to https://test.derbyramblers.org.uk/walks.html
- step 2 click on any of the walks displayed on this page. 
-This displays a popup which displays information about the walk, as part of this a map is displayed
 
You will probably see the map displayed correctly but we have three devices where the map is not displayed. However the popup uses 2 columns and if I reduce the window width it uses 1 column and it then displays the map correctly.
The devices where the map is not displayed are two HP Envy laptops running Chrome Version 94.0.4606.81, Windows 10
I also have an Samsung A51 mobile which also fails to display the map in 2 columns using Chrome. Changing it to 1 column or using a different browser works.
One of the HP laptops was working fine and then it started to have this issue, When this happened I had not knowingly updated any software and cannot see anything that changed.

I am using Leaflet 1.7

There seems to be some interaction between Chrome 2 columns and Leaflet. We have found that a map can be displayed in the 2 column state by going to map full screen and then back.
It seems to be as if the map is being transposed by the 2 column settings so that it does not appear in the correct area of the screen. Thought: is it not being transposed and is being displayed where the the map would be if the 2 column setting was not in force.

Any thoughts on how to track the problem down would be greatly appreciated.

**Comment by Falke-Design** [comment]
Try to call `map.invalidateSize()` after opening the Dialog:

```
setTimeout(function(){
    map.invalidateSize();
}, 200)
```

**Comment by ramblerswebs** [comment]
> Try to call `map.invalidateSize()` after opening the Dialog:
> 
> ```
> setTimeout(function(){
>     map.invalidateSize();
> }, 200)
> ```

thanks for the thought but just tried that it it makes no difference, Are you seeing the map displayed correctly?

**Comment by Falke-Design** [comment]
Yes on my Windows 10 it is displayed in Firefox and Chrome correctly. Can you please add a screenshot

**Comment by ramblerswebs** [comment]
If I drag the blank map up and to the right the map appears
Then I see the following div has values in the translate3D property. 
<div class="leaflet-pane leaflet-map-pane" style="transform: translate3d(0px, 0px, 0px);">

Then a fraction of a second later the transform statement returns to the above and the map vanishes.

**Comment by rapomon** [comment]
You must init the map only when is visible and the div container has a size and is not hidden, otherwise the leaflet will considered width and height 0 in the initialization.

**Comment by ramblerswebs** [comment]
I believe I am setting the map correctly.
If I change the div that contains the map, and other items, to have one column the map displays correctly. I can do this either changing the css file or by editing it in the browser
